### PR TITLE
feat!: run local `--use-task-role` flag

### PR DESCRIPTION
**Pull request by CaptainCarpensir**
Implements the second method of retrieving the task role in the credentials chain: `ecs exec`

Breaking change:
The `--use-task-role` flag will default to `true`. This is a breaking change because in the event that neither method of receiving credentials works, `copilot run local` will fail by default. To mitigate this, we introduce a recommended action displayed when `--use-task-role` fails that gives the user 2 recommendations to proceed:
1. Specify `copilot run local --use-task-role=false` to disable the feature.
2. Use a provided yaml override to allow the credential retrieving process to work.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/run_local.go [hunk 1]
```diff
@@ -666,22 +704,119 @@ func (o *runLocalOpts) taskRoleCredentials(ctx context.Context) (map[string]stri
...
+						"AWS_ACCESS_KEY_ID":     creds.AccessKeyId,
+						"AWS_SECRET_ACCESS_KEY": creds.SecretAccessKey,
+						"AWS_SESSION_TOKEN":     creds.Token,
+					}
+					return
+				}
+			case <-ctx.Done():
+				return
+			}
+			parseErr <- errors.New("all containers failed to retrieve credentials")
```
Comment (iamhopaul123) [inline internal/pkg/cli/run_local.go]: Shall we move this to the end of `case <-containersFinished:`?

File: internal/pkg/cli/run_local.go [hunk 2]
```diff
@@ -65,6 +68,10 @@ const (
 	workloadAskPrompt = "Which workload would you like to run locally?"
 )
 
+const (
+	curlContainerCredentialsCmd = "curl 169.254.170.2$AWS_CONTAINER_CREDENTIALS_RELATIVE_URI"
```
Comment (iamhopaul123) [inline internal/pkg/cli/run_local.go:74]: Can we give an example output for running this cmd here?

File: internal/pkg/cli/run_local.go [hunk 3]
```diff
@@ -1056,6 +1191,7 @@ func BuildRunLocalCmd() *cobra.Command {
 	cmd.Flags().StringVarP(&vars.envName, envFlag, envFlagShort, "", envFlagDescription)
 	cmd.Flags().StringVarP(&vars.appName, appFlag, appFlagShort, tryReadingAppName(), appFlagDescription)
 	cmd.Flags().BoolVar(&vars.watch, watchFlag, false, watchFlagDescription)
+	cmd.Flags().BoolVar(&vars.useTaskRole, useTaskRoleFlag, true, useTaskRoleFlagDescription)
```
Comment (iamhopaul123) [inline internal/pkg/cli/run_local.go:1199]: I assume the breaking change will be mitigated by https://github.com/aws/copilot-cli/pull/5529/files#diff-4648329643b6e5e2e0ee178b4b036ab8f0f5df352ba4a7e3a084eea07224121bR180? Maybe change the title to `feat!:` to indicate this PR has some breaking changes.

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/cli/run_local.go [hunk 4]
```diff
@@ -256,6 +270,32 @@ func newRunLocalOpts(vars runLocalVars) (*runLocalOpts, error) {
...
+			savedWriter.Close()
+		}
+		pipeReader, pipeWriter, err := os.Pipe()
+		if err != nil {
+			return nil, err
+		}
+		mu.Lock()
+		savedWriter = pipeWriter
+		os.Stdout = savedWriter
+		mu.Unlock()
```
Comment (KollaAdithya) [inline internal/pkg/cli/run_local.go]: Should we use a defer statement to have lock and unlock next to each other
```suggestion
		mu.Lock()
		defer mu.Unlock()
		savedWriter = pipeWriter
		os.Stdout = savedWriter
```

File: internal/pkg/cli/run_local.go [hunk 5]
```diff
@@ -666,22 +706,119 @@ func (o *runLocalOpts) taskRoleCredentials(ctx context.Context) (map[string]stri
...
+			if err != nil {
+				return nil, err
+			}
+
+			for _, container := range task.Containers {
+				wg.Add(1)
+				containerName := aws.StringValue(container.Name)
+				go func() {
+					defer wg.Done()
+					containerErr <- o.ecsExecutor.ExecuteCommand(awsecs.ExecuteCommandInput{
```
Comment (KollaAdithya) [inline internal/pkg/cli/run_local.go]: Should we wrap the error with fmt.Errorf that allows you to add extra information.

```
containerErr <- fmt.Errorf("execute command in container %s for task %s: %w", containerName, taskID, err)
```

**Review by KollaAdithya — COMMENTED** [review]
looks great!

File: internal/pkg/cli/run_local.go (hunk 5 above)
Comment (CaptainCarpensir) [reply internal/pkg/cli/run_local.go]: output is currently like:
```
ecs exec: all containers failed to retrieve credentials
execute command: TargetNotConnectedException: The execute command failed due to an internal error. Try again later.
execute command: TargetNotConnectedException: The execute command failed due to an internal error. Try again later.
```
So I'll just wrap the error with the container/task

**Review by KollaAdithya — APPROVED** [review]
🚢

**Pull request merged**